plan/logical_plan_builder.go

```diff
@@ -409,6 +411,74 @@ func (b *planBuilder) buildSelection(p LogicalPlan, where ast.ExprNode, AggMappe
 	return selection
 }
 
+// buildProjectionFieldNameForColumns builds the field name and the table name when field expression is a column reference.
+func (b *planBuilder) buildProjectionFieldNameForColumns(field *ast.SelectField, c *expression.Column) (colName model.CIStr, tblName model.CIStr) {
+	if astCol, ok := getInnerFromParentheses(field.Expr).(*ast.ColumnNameExpr); ok {
+		colName = astCol.Name.Name
+		tblName = astCol.Name.Table
+	} else {
+		colName = c.ColName
+		tblName = c.TblName
+	}
+	return
+}
+
+// buildProjectionFieldNameForExpressions builds the field name when field expression is a normal expression.
+func (b *planBuilder) buildProjectionFieldNameForExpressions(field *ast.SelectField) (colName model.CIStr) {
+	if agg, ok := field.Expr.(*ast.AggregateFuncExpr); ok && agg.F == ast.AggFuncFirstRow {
+		// When the query is select t.a from t group by a; The Column Name should be a but not t.a;
+		colName = agg.Args[0].(*ast.ColumnNameExpr).Name.Name
```

Comment:
return colName here `return agg.Args[0].(*ast.ColumnNameExpr).Name.Name` and remove the `else` statement next line